java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshot.java

```diff
@@ -38,6 +38,13 @@ BarrageSnapshot snapshot(TableSpec tableSpec, BarrageSnapshotOptions options)
         BarrageSnapshot snapshot(TableHandle tableHandle, BarrageSnapshotOptions options);
     }
 
+    /**
+     * This call will return false until all rows for the result table are available or the snapshot failed.
+     *
+     * @return true when all rows for the result table are available or the snapshot failed, false otherwise
+     */
+    boolean isCompleted();
```

Comment:
Maybe we should return an enum here? Success or failure? Or maybe we should just be returning a `Future` from `partialTable` and `entireTable` anyway. That would avoid letting the user get a reference before it's safe to do so.